src/middleware/immer.ts

```diff
@@ -57,9 +57,19 @@ const immerImpl: ImmerImpl = (initializer) => (set, get, store) => {
   type T = ReturnType<typeof initializer>
 
   store.setState = (updater, replace, ...a) => {
-    const nextState = (
-      typeof updater === 'function' ? produce(updater as any) : updater
-    ) as ((s: T) => T) | T | Partial<T>
+    if (replace === false) {
+      throw new Error(
+        '`replace` must be `true` (or nullish) when using the `immer` middleware.',
+      )
+    }
+    // We always want to `replace` with `immer`.
+    replace = true
+    if (typeof updater !== 'function') {
+      throw new Error(
+        'You must pass a function to `setState` when using the `immer` middleware.',
+      )
+    }
+    const nextState = produce(updater as any) as (s: T) => T
 
     return set(nextState as any, replace, ...a)
```

Comment:
then set `replace` as `true` by default

```suggestion
    const nextState = produce(updater as any) as (s: T) => T

    return set(nextState as any, true, ...a)
```

Reply:
Just to be sure: also remove the check on `updater`?

Reply:
I applied your suggestions and pushed a new commit but it breaks the typing, but I'm not sure why.

It's not happy with this line:
```typescript
return set(nextState as any, /* replace */ true, ...a)
```
(`Expected 1-2 arguments, but got 3.`)

If I put back the `replace` in this line the error goes away:
```typescript
 store.setState = (updater, replace, ...a) => {
```

But I'm really not sure why.

Reply:
Let me look into that quickly